libraries/APM_Control/AR_AttitudeControl.cpp

```diff
@@ -802,9 +802,8 @@ float AR_AttitudeControl::get_desired_speed() const
 // get acceleration limited desired speed
 float AR_AttitudeControl::get_desired_speed_accel_limited(float desired_speed, float dt) const
 {
-    // return input value if no recent calls to speed controller
     // apply no limiting when ATC_ACCEL_MAX is set to zero
-    if (!speed_control_active() || !is_positive(_throttle_accel_max)) {
```

Comment:
Have you tested what happens though the engaging the speed controller at a very different speed that it was previously active at?

For example, drive the vehicle at high speed in Manual mode and then switch to Acro... does the desired_speed start from zero or from the current speed?  Conversely drive the vehicle in Acro at high speed and then switch to manual, slow down and switch back to Acro.